docs/enhancements/image-service-cloud-authentication.md

```diff
@@ -0,0 +1,164 @@
+
+## Proposal
+
+The new authentication mechanism will be a simple secret and expiration date
+both of which are tied to an infraEnv. This secret will be included as a
+parameter in the image download URL and assisted service authentication will
+pass if the secret in the URL matches the one currently in the infraEnv record
+and the request is processed before the expiration time. A new secret will be
+generated any time a user updates the infraEnv in a way that would change the
+image content. Additionally a new endpoint will be required to force a new
+secret to be created in the event that the current one has expired and no new
+changes to the infraEnv are necessary.
+
+The secret will be a random URL-safe string. In particular it will not be a JWT
+and it will not attempt to encode any information about a user, resources, or its
+own expiration (as a JWT typically does). This approach was chosen to simplify the
+proposal and to eliminate the extra effort and risk associated with managing
+encryption keys for a publicly accessible application. This approach also is
+no less secure than the previous solution where a pre-signed S3 URL could be
+downloaded by anyone with access to the URL itself.
+
+### User Stories
+
+#### Story 1
+
+As a user of assisted service via console.redhat.com, I want a simple download
+URL to access the discovery ISO. This URL should be usable with standard download
+utilities (i.e. `curl`, `wget`) without the need for request headers or additional tools
+like the `ocm` CLI.
+
+#### Story 2
+
+As a user of assisted service via console.redhat.com, I want my sensitive credentials
+and infrastructure information contained within the discovery ISO to be kept secure.
+
+#### Story 3
+
+As a user of assisted service via console.redhat.com, I want the discovery ISO download
+URL to be usable by anyone I choose to give it to without disclosing my personal Red Hat
+account credentials or API tokens.
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+Assisted service will implement a new security definition to serve this purpose.
+The existing definitions already have well-defined use-cases and scopes that do not
+fully match the requirements of this proposal. Specifically the existing urlAuth security
+definition applies to more endpoints than we would want to expose using just a secret.
+
+This security definition will apply only to the following endpoints:
+
+* `/v2/infra-envs/{infra_env_id}/downloads/files`
+* `/v2/infra-envs/{infra_env_id}/downloads/minimal-initrd`
+
+These are the endpoints that the image service uses to fetch image customization information
+and are the only ones that are needed for this process.
+
+The secret and expiration date will be stored as columns in the infraEnv database table
+and will not be accessible through the infraEnv API. The expiration will remain the same (4 hours)
+
+A new API endpoint will be added:
+
+* `/v2/infra-envs/{infra_env_id}/downloads/refresh-image-secret`
```

Comment:
maybe it should be called image-secret if we have a CRUD for it then refresh doesn't sounds right, @avishayt what do you think?

Reply:
I'm not sure we'll have a CRUD. Maybe the endpoint should update the secret and return the URL to the image?

So something like `GET /v2/infra-envs/{infra_env_id}/downloads/image-url` would work I suppose.
But that doesn't make a ton of sense given that the image url is already in the infra env as a property.